R/clean_redcap.R

```diff
@@ -163,6 +168,7 @@ distill_nonrepeat_table <- function(form_name,
       any_of(c("redcap_event", "redcap_arm", "redcap_form_instance", "redcap_event_instance")),
       .after = all_of(my_record_id)
     ) %>%
+    relocate(any_of("redcap_data_access_group"), .after = all_of(my_record_id)) %>%
```

Comment:
So `redcap_data_access_group` will come after `record_id`? Hm, I'm wondering if that's the best place. My gut feeling is that pushing this all the way to the right would be better, into the proximity of the survey identifiers?

Reply:
In REDCapR it comes after the `record_id`, `redcap_repeat_instance`, `redcap_repeat_instrument`, `redcap_repeat_instance` columns:

```r
> redcapr_out_dag <- REDCapR::redcap_read_oneshot(redcap_uri, token, export_data_access_groups = TRUE)$data %>% tibble()
6 records and 8 columns were read from REDCap in 1.1 seconds.  The http status code was 200.                                           
> redcapr_out_dag
# A tibble: 6 × 8
  record_id redcap_repeat_instrument redcap_repeat_instance redcap_data_access_group
      <dbl> <chr>                                     <dbl> <chr>                   
1         1 NA                                           NA dag1                    
2         1 repeat_form_1                                 1 dag1                    
3         1 repeat_form_1                                 2 dag1                    
4         2 NA                                           NA dag2                    
5         3 NA                                           NA dag3                    
6         4 NA                                           NA NA                      
# ℹ 4 more variables: non_repeat_form_1_txt <lgl>, non_repeat_form_1_complete <dbl>,
#   repeat_form_1_txt <dbl>, repeat_form_1_complete <dbl>
```

Personally after working with REDCapR a lot prior to our package and in the context of DAGs, I'd prefer to see it in a similar place and not at the end of the dataset. I think this makes sense since from an analyst point of view this an additional identifier to work off of, even if its not a part of the composite key.

Reply:
Thanks for the additional info. I would argue that we are already taking some liberties with changing the non-data column structure to make it more logical. As of now, REDCapTidieR logic is that you have the composite key, then data columns, and then row-level metadata columns. I think DAG is a row-level metadata column akin to the survey identifier or form completion question. Might be worth discussing at standup.

Reply:
Noting from today's meeting, after discussion opting to keep DAG after composite key identifiers. Sent a message to Paul as well for general feedback.